File: base_java/src/main/java/com/hesheng1024/base_java/ContextHolder.java
package com.hesheng1024.base_java;

import android.annotation.SuppressLint;
import android.app.Application;
import android.content.Context;

import androidx.annotation.NonNull;

/**
 * @author hesheng1024
 * @date 2020/5/6 18:45
 */
public final class ContextHolder {

    private ContextHolder() {
    }

    private static Application mApplication;
    @SuppressLint("StaticFieldLeak")
    private static Context mContext;

    static void init(@NonNull Application application, @NonNull Context context) {
        mApplication = application;
        mContext = context;
    }

    public static Application getApplication() {
        return mApplication;
    }

    public static Context getContext() {
        return mContext;
    }
}

File: base_kotlin/src/main/java/com/hesheng1024/base_kotlin/ContextHolder.kt
package com.hesheng1024.base.utils

import android.app.Application
import android.content.Context

/**
 *
 * @author hesheng1024
 * @date 2020/2/7 11:50
 */
object ContextHolder {
    private lateinit var sContext: Context
    private lateinit var sApplication: Application

    fun holdMainApplication(application: Application) {
        sApplication = application
    }

    fun holdMainContext(context: Context) {
        sContext = context
    }

    fun getMainContext(): Context = sContext

    fun getMainApplication(): Application = sApplication
}